pyboy/gymboy.py

```diff
@@ -0,0 +1,165 @@
+            If true, the agent is allowed to inject multiple inputs at the same time. Caution, this also means that the action_space is way bigger (n -> 2^n)!
+
+    Attributes
+    ----------
+
+        game: `pyboy.PyBoy`
+            The PyBoy game instance over which the environment is built.
+        
+        observation_type: str
+            Define what the agent will be able to see :
+                - 'raw' gives the raw RGB pixels color
+                - 'tiles' gives the id of the sprites in 8x8 pixel zones of the game_area defined by the game_wrapper (Only useful in grid-based games).
+
+        buttons_press_mode: str
+            Define how the agent will interact with button inputs:
+                - 'press' the agent will only press inputs for 1 frame an then release it.
+                - 'toggle' the agent will toggle inputs, first time it press and second time it release.
+                - 'all' the agent have acces to all inputs, press and release are separated.
+        
+        game_wrapper: `pyboy.plugins.base_plugin.PyBoyGameWrapper`
+            The game_wrapper of the PyBoy game instance over which the environment is built.
+
+        action_space: Gym space
+            The action space of the environment.
+
+        observation_space: Gym space
+            The observation space of the environment (depends of observation_type).
+        
+        last_fitness: float or int
+            The last observed fitness, used to compute the reward at each step
+
+        actions: list
+            The list of input IDs of allowed input for the agent (depends of buttons_press_mode).
+    
+    """
+
+    def __init__(self, game, observation_type='tiles', buttons_press_mode='toggle', simultaneous_actions=False):
+        # Build game
+        self.game = game
+        if str(type(game)) != "<class 'pyboy.pyboy.PyBoy'>":
+            raise TypeError("game must be a Pyboy object")
+        
+        # Build game_wrapper
+        self.game_wrapper = game.game_wrapper()
+        if self.game_wrapper is None:
+            raise ValueError("You need to build a game_wrapper to use this function. Otherwise there is no way to build a reward function automaticaly (for now).")
+        self.last_fitness = self.game_wrapper.fitness
+
+        # Building the action_space
+        self._DO_NOTHING = -1
+        self._buttons = list(range(1, 9)) # UP DOWN RIGHT LEFT A B SELECT START
+        self._button_is_pressed = {button:False for button in self._buttons}
+        
+        self._buttons_release = list(range(9, 17)) # UP DOWN RIGHT LEFT A B SELECT START
+        self._release_button = {button:r_button for button, r_button in zip(self._buttons, self._buttons_release)}
+        
+        self.actions = [self._DO_NOTHING] + self._buttons
+        if buttons_press_mode == 'all':
+            self.actions += self._buttons_release
+        elif not (buttons_press_mode == 'press' or buttons_press_mode == 'toggle'):
```

Comment:
Can also be written `elif buttons_press_mode not in ['press', 'toggle']:`

Reply:
cleaner, i like it